Docs: Explain the runtime override of active series matchers (#1868)
grafana/mimir: docs/sources/operators-guide/configuring/configuring-custom-trackers.md

Series with metric name `cortex_ingester_active_series_custom_tracker` have two labels applied: `name` and `user`. The value of the `name` label is the name of the custom tracker specified in the configuration. The value of the `user` label is the tenant-id for which the series count applies.

> To illustrate, assume that two custom trackers are configured as in the yaml shown above, and that your Grafana Mimir cluster has two tenants: `tenant_1` and `tenant_with_only_prod_metrics`. Assume that `tenant_with_only_prod_metrics` has 3 series with labels that match the pattern `{namespace=~"prod-.*"}` and none that match `{namespace=~"dev-.*"}` and that `tenant_1` has 5 series that match the pattern `{namespace=~"dev-.*"}` and 10 series that match the pattern `{namespace=~"prod-.*"}`.

Reviewer: ```suggestion
To illustrate this, assume that two custom trackers are configured as in the preceding YAML snippet, and that your Grafana Mimir cluster has two tenants: `tenant_1` and `tenant_with_only_prod_metrics`. Assume that `tenant_with_only_prod_metrics` has three series with labels that match the pattern `{namespace=~"prod-.*"}` and none that match the patten `{namespace=~"dev-.*"}`. Also assume that `tenant_1` has five series that match the pattern `{namespace=~"dev-.*"}` and 10 series that match the pattern `{namespace=~"prod-.*"}`.
```
* "to illustrate" is a transitive verb, meaning that it takes on object.
* Reduce the length of a complete sentence so it is easier to read.
* Spell out numbers between zero and nine.

Author: I am not going to add this to each suggestion to avoid noise, but: this is pure gold, thank you so much for reviewing! :)

> You should see the following output when the `/metrics` endpoint for the ingester component is scraped:

Reviewer: ```suggestion
In this example, the following output appears when the `/metrics` endpoint for the ingester component is scraped:
```
Revert this because:
* The term _should_ is ambiguous. Rephrase it so that the advice is unambigous and accurate.

```
cortex_ingester_active_series_custom_tracker{name="dev", user="tenant_1"}                         5
cortex_ingester_active_series_custom_tracker{name="prod", user="tenant_1"}                       10
cortex_ingester_active_series_custom_tracker{name="prod", user="tenant_with_only_prod_metrics"}   3
```

> Starting with Mimir version 2.1 default configuration as described above can be overridden for specific tenants in the [runtime configuration]({{< relref "./about-runtime-configuration.md" >}}).
> 

Reviewer: ```suggestion
For specific tenants, you can override the default configuration as previously described. To do so, edit the  [runtime configuration]({{< relref "./about-runtime-configuration.md" >}}).

```
* The documentation is already versioned, so this version number need not be embedded in the content itself. It could make sense to add a version number in a migration guide, if that is needed. Is it?
* Remove directional language because text does not always flow above, below, left, or right; prefer _preceding_ or _that follows_ as an alternative to each.
* Replace passive voice with active voice.
* Simplify the entire sentence by breaking it up into two; there are many details listed in a single sentence that make it too compact to read, especially for non-native speakers.

> The example below shows how you could override the active series custom trackers configuration for tenant `tenant_with_only_prod_metrics` to track two interesting services instead of the default matchers.

Reviewer: ```suggestion
You can override the active series custom trackers’ configuration for the tenant `tenant_with_only_prod_metrics` to track two services instead of the default matchers. See the following example:
```
* Replace directional language (below) with _that follows_; see preceding comment.
* The term _could_ is ambiguous. Rephrase it so that the advice is unambiguous and accurate.
* I assume there is more than one tracker; if there is, use plural possession (trackers’). If there is only one, use singular possession (tracker’s).
* The term _interesting_ is subjective, so remove it.
* Do you mean _matches_ or _matchers_?

Reviewer: Why is this service interesting? What is it doing? Can we make this example a bit more concrete?

Reviewer 2: we can just go with `service=service1` and `service=service2` @osg-grafana . wdyt? the idea would be 2 services that are unique to the prod metrics tenant that we care about their series count.

```
overrides:
  tenant_with_only_prod_metrics:
    active_series_custom_trackers:

>       interesting-service: '{service=~"interesting-.*"}'
>       also-interesting-service: '{service=~"also-interesting-.*"}'

Reviewer: ```suggestion
      interesting-service: '{service=~"interesting-.*"}'
      also-interesting-service: '{service=~"also-interesting-.*"}'
```
See previous comment about "interesting".

Author: I am going to go with service1 and service2.

```

> After adding this override, and assuming there are 1 and 2 matching series for `interesting-service` and `also-interesting-service`, respectively, the output at `/metrics` would change to:

Reviewer: ```suggestion
After adding this override, and assuming that there is one matching series for `interesting-service` and two matching series for `also-interesting-service`, the output at `/metrics` changes:
```
I removed the ending preposition so that the sentence is self-contained.

```
cortex_ingester_active_series_custom_tracker{name="dev", user="tenant_1"}                                           5
cortex_ingester_active_series_custom_tracker{name="prod", user="tenant_1"}                                         10
cortex_ingester_active_series_custom_tracker{name="interesting-service", user="tenant_with_only_prod_metrics"}      1
cortex_ingester_active_series_custom_tracker{name="also-interesting-service", user="tenant_with_only_prod_metrics"} 2
```

> For detailed information how to set up runtime overrides, refer to [runtime configuration]({{< relref "./about-runtime-configuration.md" >}}).

Reviewer: ```suggestion
To set up runtime overrides, refer to [runtime configuration]({{< relref "./about-runtime-configuration.md" >}}).
```
